src/AuditableBehavior.php

```diff
@@ -4,12 +4,22 @@ class AuditableBehavior extends Behavior
 {
     public function objectAttributes($builder)
     {
-        return 'protected $originalFieldValues = null;';
+		return "protected \$originalFieldValues = null; \n protected \$wasNew = false; \n protected \$wasModified = false;";
+    }
+
+    public function preSave()
+    {   
+        return "\$this->wasNew = \$this->isNew();";
+    }
+
+    public function preUpdate()
+    {
+        return "\$this->wasModified = \$this->isModified();";
     }
     
     public function postHydrate()
     {
-        return '$this->originalFieldValues = $row;';
+        return "\$this->wasNew = \$this->getWasNew(); \n \$this->wasModified = \$this->getWasModified(); \n \$this->originalFieldValues = \$row;";
```

Comment:
Can you explain this line? Why are you setting the value of $this->wasNew to a method that will return the same value? Am I missing something here? Shouldn't  this be $this->wasNew = $this->isNew().